File: segundos 4 lenguajes/go/JugadorExperto.go
package main

type JugadorExperto struct {
	Nombre string
	Nivel  int
}

func NewJugadorExperto() JugadorExperto {
	return JugadorExperto{"Experto", 20}
}

func NewJugadorExpertoParams(nombre string, nivel int) JugadorExperto {
	return JugadorExperto{nombre, nivel}
}

func NewJugadorExpertoNombre(nombre string) JugadorExperto {
	return JugadorExperto{nombre, 20}
}

func NewJugadorExpertoCopy(j JugadorExperto) JugadorExperto {
	return JugadorExperto{j.Nombre, j.Nivel}
}

File: segundos 4 lenguajes/Java/JugadorExperto.java
package Java;
class JugadorExperto {
    String nombre;
    int nivel;

    public JugadorExperto() {
        nombre = "Experto";
        nivel = 20;
    }

    public JugadorExperto(String nombre, int nivel) {
        this.nombre = nombre;
        this.nivel = nivel;
    }

    public JugadorExperto(String nombre) {
        this.nombre = nombre;
        this.nivel = 20;
    }

    public JugadorExperto(JugadorExperto j) {
        this.nombre = j.nombre;
        this.nivel = j.nivel;
    }
}
